lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/Task/TaskClientCodeException.cs

```diff
@@ -16,15 +16,41 @@
 // under the License.
 
 using System;
-using Org.Apache.REEF.Tang.Interface;
+using System.Runtime.Serialization;
+using System.Security.Permissions;
 
 namespace Org.Apache.REEF.Common.Runtime.Evaluator.Task
 {
-    internal sealed class TaskClientCodeException : Exception
+    /// <summary>
+    /// An Exception thrown when Task operations (Start, Stop, Suspend) fail.
+    /// </summary>
+    [Serializable]
+    public sealed class TaskClientCodeException : Exception
     {
-        private readonly string _taskId;
+        private const string TaskIdStr = "TaskId";
+        private const string ContextIdStr = "ContextId";
 
         private readonly string _contextId;
+        private readonly string _taskId;
+
+        internal static TaskClientCodeException Create(
```

Comment:
Because we want the second constructor doesn't expect an actual `Exception`, while we implicitly want the `Exception` to be included in the message. This will delegate all the responsibility of calling `Exception.ToString()` on the caller, which is error-prone.